app/models/upload.rb

```diff
@@ -54,6 +54,28 @@ def self.last_uploads
     Upload.select('DISTINCT ON("csv_type") *').where(ok: true).order(csv_type: :asc).order(updated_at: :desc)
   end
 
+  def self.last_uploads_rows
+    uploads = Upload.last_uploads
+
+    uploads_hash = {}
+    uploads.each do |r|
+      uploads_hash[r.csv_type] = r
+    end
```

Comment:
Instead of instantiating a big hash, we can simplify this by just doing something like:
```
upload_csv_types = uploads.map(&:csv_type)

# then on line 67:
next if upload_csv_types.include?(klass.name)
```